### ISSUE TITLE
Lint for non-immutable generated columns

### ISSUE THREAD

**Issue by darora** [supabase/splinter#58]
Generated columns in Postgres are required to only use immutable functions, but this does not seem to be enforced at creation time. It can, however, cause the upgrade process to fail.

**Comment by olirice** [comment]
> this does not seem to be enforced at creation time

I tried 
```sql
create table abc (
  id int primary key,
  x timestamp generated always as ( now() ) stored
);
```
on 15.1 and 16.2 and got a 
```
ERROR:  42P17: generation expression is not immutable
```
in both cases

@darora do you have a reproducible way to get into the bad state that I can use for testing?

[relevant (internal) slack thread](https://supabase.slack.com/archives/C01D6TWFFFW/p1713459800807569)

**Comment by olirice** [comment]
closing due to inactivity

**Comment by darora** [comment]
I do not, unfortunately. Let's close this unless we run into the issue again. We've only seen it the one time.
